### PR TITLE
Metadata as Buffer. New Metadata APIs

### PR DESCRIPTION
**Pull request by marcopiraccini**
Here metadata values are`Buffer`.
Also slightly changed the metadata API, see the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,12 +79,26 @@ sdswim.on('peerDown', node => {
...
+Change a single metadata:
+
+```
+node.add('myKey', Buffer.from('myValue'))
+```
+
+Remove a single metadata:
+
+```
+node.node.remove('test2')
```
Comment (mcollina) [inline README.md]: this is not nice, why `node.node.remove` and not `node.remove`?

Comment (marcopiraccini) [inline README.md]: It's an error in the README :)

File: lib/metadata.js [hunk 2]
```diff
@@ -73,13 +80,38 @@ class Metadata extends EventEmitter {
...
+  /**
+   * Add a single entry
+   */
+  add (key, value) {
+    if (!isBuffer(value)) {
+      throw new Error('value must be a Buffer')
     }
+    this.entries.push({ key, value })
+    this.version++
+    this.notifyToAll()
```
Comment (mcollina) [inline lib/metadata.js]: Maybe we should check for duplicates?

**Review by mcollina — APPROVED** [review]
LGTM

**Pull request merged**